examples/keras_recipes/tensorflow_numpy_models.py

```diff
@@ -0,0 +1,340 @@
+            self.block_weights.append(
+                self.add_weight(
+                    shape=(current_shape, block), trainable=True, name=f"block-{i}"
+                )
+            )
+            self.biases.append(
+                self.add_weight(shape=(block,), trainable=True, name=f"bias-{i}")
+            )
+            current_shape = block
+
+        self.linear_layer = self.add_weight(
+            shape=(current_shape, 1), name="linear_projector", trainable=True
+        )
+
+    def call(self, inputs):
+        activations = inputs
+        for w, b in zip(self.block_weights, self.biases):
+            activations = tnp.matmul(activations, w) + b
+            # ReLu activation function
+            activations = tnp.maximum(activations, 0)
+
+        return tnp.matmul(activations, self.linear_layer)
+
+
+"""
+Just like with any other Keras model we can utilize any supported optimizer, loss,
+metrics or callbacks that we want.
+"""
+
+"""
+Let's see how the model performs!
+"""
+
+model = TNPForwardFeedRegressionNetwork(blocks=[3, 3])
+model.compile(
+    optimizer="adam",
+    loss="mean_squared_error",
+    metrics=[keras.metrics.MeanAbsolutePercentageError()],
+)
+evaluate_model(model)
+
+"""
+Great!  Our model seems to be effectively learning to solve the problem at hand.
+"""
+
+"""
+# Implementing a Keras Layer Based Model with TNP
+If desired, `tnp` can also be used in layer oriented Keras code structure.  Let's
+implement the same model, but using a layered approach!
+"""
+
+
+class DenseTNP(keras.layers.Layer):
+    def __init__(self, units):
+        super().__init__()
+        self.units = units
+
+    def build(self, input_shape):
+        self.w = self.add_weight(
+            name="weights",
+            shape=(input_shape[1], self.units),
+            initializer="random_normal",
+            trainable=True,
+        )
+        self.bias = self.add_weight(
+            name="bias",
+            shape=(self.units,),
+            initializer="random_normal",
+            trainable=True,
+        )
+
+    def call(self, inputs):
+        return tnp.maximum(tnp.matmul(inputs, self.w) + self.bias, 0)
+
+
+class LinearProjector(keras.layers.Layer):
+    def build(self, input_shape):
+        self.w = self.add_weight(
+            name="weights",
+            shape=(input_shape[1], 1),
+            initializer="random_normal",
+            trainable=True,
+        )
+        self.bias = self.add_weight(
+            name="bias", shape=(1,), initializer="random_normal", trainable=True
+        )
+
+    def call(self, inputs):
+        return tnp.matmul(inputs, self.w) + self.bias
+
+
+from keras.models import Sequential
+
+
+def create_layered_tnp_model():
+    model = Sequential()
+    model.add(DenseTNP(3))
+    model.add(DenseTNP(3))
+    model.add(LinearProjector())
+    return model
+
+
+model = create_layered_tnp_model()
+model.compile(
+    optimizer="adam",
+    loss="mean_squared_error",
```

Comment:
Random though: maybe it would be useful to show how to define the loss in NumPy too

Reply:
Went ahead and added a section on this above 👍